drivers/input/input_cst816s.c

```diff
@@ -292,16 +320,123 @@ static int cst816s_init(const struct device *dev)
 	return ret;
 }
 
-#define CST816S_DEFINE(index)                                                                      \
-	static const struct cst816s_config cst816s_config_##index = {                              \
-		.i2c = I2C_DT_SPEC_INST_GET(index),                                                \
-		COND_CODE_1(CONFIG_INPUT_CST816S_INTERRUPT,                                        \
-			    (.int_gpio = GPIO_DT_SPEC_INST_GET(index, irq_gpios),), ())           \
-			.rst_gpio = GPIO_DT_SPEC_INST_GET_OR(index, rst_gpios, {}),                \
-	};                                                                                         \
-	static struct cst816s_data cst816s_data_##index;                                           \
-	DEVICE_DT_INST_DEFINE(index, cst816s_init, NULL, &cst816s_data_##index,                    \
-			      &cst816s_config_##index, POST_KERNEL, CONFIG_INPUT_INIT_PRIORITY,    \
-			      NULL);
+#ifdef CONFIG_PM_DEVICE
+static int cst816s_apply_profile(const struct cst816s_config *cfg)
+{
+	int ret;
+
+	ret = i2c_burst_write_dt(&cfg->i2c, CST816S_REG_AUTOSLEEP_TIME, (uint8_t *)&cfg->lp_profile,
+				 sizeof(cfg->lp_profile));
+	if (ret) {
+		LOG_WRN("Write power profile failed (%d)", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int cst816s_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	int ret;
+	const struct cst816s_config *cfg = dev->config;
+
+	/* For some reason the CST816S does not respond to I2C commands after we use the standby
+	 * profile. Workaround for now is to just always reset it before we change power modes.
+	 */
+	cst816s_chip_reset(dev);
+	ret = cst816s_chip_init(dev);
+	if (ret < 0) {
+		LOG_ERR("Chip init failed during PM action (%d)", ret);
+		return ret;
+	}
+
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+	/* For PM TURN_ON means device is in suspend mode, hence apply suspend profile. */
+	case PM_DEVICE_ACTION_TURN_ON:
+		ret = cst816s_apply_profile(cfg);
+		if (ret < 0) {
+			LOG_WRN("Could not apply suspend profile! (%d)", ret);
```

Comment:
```suggestion
		if (ret < 0) {
			LOG_WRN("Could not apply suspend profile (%d)", ret);
```

Sorry I know this may be the most nitpicky comment ever, it's odd to have the extra emphasis on this, does not strike me as good use of flash memory :-)